plugins/woocommerce/src/Internal/Updates/BatchProcessingController.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Controller to manage updates of `BatchProcessor` type. Trigger the `enqueue_processor` method to start a update, and it will manage rest of the things.
+ */
+
+namespace Automattic\WooCommerce\Internal\Updates;
+
+use Automattic\WooCommerce\DataBase\BatchProcessor;
+
+/**
+ * Class BatchProcessingController
+ *
+ * @package Automattic\WooCommerce\Internal\Updates.
+ */
+class BatchProcessingController {
+
+	const PENDING_PROCESS_OPTION_NAME = 'wc_pending_action_updates';
+	const CONTROLLER_CRON_NAME = 'wc_schedule_pending_updates';
+	const SINGLE_PROCESS_MIGRATION_ACTION = 'wc_run_action_update';
+	const ACTION_GROUP = 'wc_updates';
+
+	/**
+	 * BatchProcessingController constructor.
+	 *
+	 * Schedules necessary actions to process batches.
+	 */
+	public function __construct() {
+		add_action( self::CONTROLLER_CRON_NAME, array( $this, 'schedule_processes' ) );
+		add_action( self::SINGLE_PROCESS_MIGRATION_ACTION, array( $this, 'process_single_update' ), 10, 2 );
+	}
+
+	/**
+	 * Starts an update.
+	 *
+	 * @param string $updater_class_name Fully qualified class name of an updater, must be child class of `BatchProcessor`.
+	 */
+	public function enqueue_processor( string $updater_class_name ) {
```

Comment:
Parameter name change suggestion:

```suggestion
	public function enqueue_processor( string $processor_class_name ) {
```

Reply:
👍  Added in https://github.com/woocommerce/woocommerce/pull/33233/files#diff-8989ad7f50c6e63bf0ff83b5df17d72121166df76e16875ec985062b8ca740d5R35